client/src/redux/actions/twitterActions.js

```diff
@@ -127,16 +133,144 @@ export const tweetNewMessage = ({ message }) => async (dispatch) => {
       },
     });
   } catch (err) {
-    if (err.response.data) {
-      dispatch({
-        type: SET_ERRORS,
-        payload: {
-          errors: err.response.data.errors,
-        },
-      });
-      dispatch({
-        type: TWITTER_RESET_PROFILE_INFO,
-      });
+    dispatch(handleError({ error: err }));
+  }
+};
+
+export const updateStreams = ({ streams }) => async (dispatch) => {
+  await dispatch({
+    type: TWITTER_UPDATE_STREAMS,
+    payload: {
+      streams: streams,
+    },
+  });
+};
+
+export const updateUserStreamsBackend = ({ streams }) => async (dispatch) => {
+  try {
+    const updatedstreams = [...streams].map((stream) => ({
```

Comment:
It's safe to just do `streams.map()` here, without spreading the array. `map()` leaves the array unmodified and creates a new one.

Reply:
Fixed. Also fixed this for loadUserStreams Action as it was the same issue